recipes/fmtlog/all/conanfile.py

```diff
@@ -0,0 +1,119 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.layout import basic_layout
+import os
+
+
+required_conan_version = ">=1.52.0"
+
+
+class PackageConan(ConanFile):
+    name = "fmtlog"
+    description = "fmtlog is a performant fmtlib-style logging library with latency in nanoseconds."
+    license = "MIT license"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/MengRao/fmtlog"
+    topics = ("logging", "low-latency", "topic3") # no "conan" and project name in topics
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "header_only": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "header_only": False,
+    }
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "Visual Studio": "15.7",
+            "msvc": "191",
+            "clang": "7",
+            "apple-clang": "10",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        if self.options.header_only:
+            basic_layout(self, src_folder="src")
+        else:
+            cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("fmt/9.1.0")
+
+    def package_id(self):
+        if self.options.header_only:
+            self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
```

Comment:
```suggestion
        if self.info.settings.get_safe("compiler.cppstd"):
```
Because when header-only=True this recipe will build nothing, only copy headers. So it matters for who is installing the package. On the other hand, when building sources, cppstd will be there if declared.

Reply:
Now I understand.
Reference:
https://github.com/conan-io/conan-center-index/pull/13170#discussion_r981407617.